docs/HowToGuides/GettingStarted.md

```diff
@@ -0,0 +1,567 @@
+     systemtap-sdt-dev     \
+     tzdata                \
+     uuid-dev
+   sudo snap install sccache --candidate --classic
+   ```
+
+   For Ubuntu 20.04 LTS, you can run the same command
+   but with `libpython2-dev` instead of `libpython-dev`.
+
+   **Note:** LLDB currently requires at least `swig-1.3.40` but will
+   successfully build with version 2 shipped with Ubuntu.
+
+## Building the project for the first time
+
+### Spot check dependencies
+
+* Run `cmake --version`: This should be 3.18.1 or higher for macOS.
+* Run `python3 --version`: Check that this succeeds.
+* Run `ninja --version`: Check that this succeeds.
+* Run `sccache --version`: Check that this succeeds.
+
+### The roles of different tools
+
+At this point, it is worthwhile to pause for a moment
+to understand what the different tools do:
+
+1. CMake is a cross-platform build system for C and C++.
+   It forms the core infrastructure used to _configure_ builds of
+   Swift and its companion projects.
+2. Ninja is a low-level build system that can be used to _build_ the project,
+   as an alternative to Xcode's build system. Ninja is somewhat faster,
+   especially for incremental builds, and supports more build environments.
+3. Sccache is a caching tool:
+   If you ever delete your build directory and rebuild from scratch
+   (i.e. do a "clean build"), Sccache can accelerate the new build
+   significantly. There are few things more satisfying than seeing Sccache
+   cut through build times.
+4. `utils/update-checkout` is a script to help you work with all the individual
+   git repositories together, instead of manually cloning/updating each one.
+5. `utils/build-script` (we will introduce this shortly)
+   is a high-level automation script that handles configuration (via CMake),
+   building (via Ninja or Xcode), caching (via Sccache), running tests and more.
+
+> **Pro Tip**: Most tools support `--help` flags describing the options they
+> support. Additionally, both Clang and the Swift compiler have hidden flags
+> (`clang --help-hidden`/`swiftc --help-hidden`) and frontend flags
+> (`clang -cc1 --help`/`swiftc -frontend --help`) and the Swift compiler
+> even has hidden frontend flags (`swiftc -frontend --help-hidden`). Sneaky!
+
+Phew, that's a lot to digest! Now let's proceed to the actual build itself!
+
+### The actual build
+
+1. Make sure you have Sccache running.
+   ```sh
+   sccache --start-server
+   ```
+   (Optional) Sccache defaults to a cache size of 10GB, which is relatively
+   small compared to build artifacts. You can bump it up, say by setting
+   `export SCCACHE_CACHE_SIZE="50G"` in your dotfile(s). For more details,
+   see the [Sccache README][Sccache].
+2. Decide if you would like to build the toolchain using Ninja or using Xcode.
+   - If you use an editor other than Xcode and/or you want somewhat faster builds,
+     go with Ninja.
+   - If you are comfortable with using Xcode and would prefer to use it,
+     go with Xcode.
```

Comment:
Perhaps it may be worth noting that you can use both (Xcode for editing, Ninja for building)?

Reply:
Yeah, I thought about bringing this up, but I struggled on how to present it. More generally, you'll notice that I've tried to minimize decision-making in favor of selecting what I think are reasonable defaults.

While managing two builds is a reasonable workflow, it also has its downsides in terms of making sure they don't go out of sync. Describing the workflow would require at least 1-2 paragraphs.

Maybe I could split it out into a separate subsection and link that here.

Reply:
Last time I checked, `build-script --xcode` didn't actually have a way to run tests. Has that changed?

(If it hasn't, I think that gives you a way to simplify your description: mostly talk about Ninja as if it were the only option, then separately present Xcode as an additional option you might use for editing and debugging alongside your Ninja build for testing.)

Reply:
There is a `check-swift` target which runs all the tests. https://forums.swift.org/t/stuck-on-compiling/39711/9 . I didn't have success in getting that target to build, but I figured @Jumhyn had, so I mentioned it.

Reply:
Yeah, I haven't experimented with the litany of `check-swift*` targets, but I've been able to successfully build/run `check-swift-all` which works for running all the tests (excepting some failures which reliably appear in Xcode but not elsewhere).

Reply:
I've created a separate sub-sub-section describing the dual build workflow and linked that here.